### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by yuxiang-zhang**
### Description

Closes #7472 #7492 #7494 #7506 #7508 #7510 #7514 #7527 #7528 #7529 #7531 #7532 #7533 #7534 #7535 #7536 #7537 #7544 #7545 #7546 #7547 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/kubernetes/client.go [hunk 1]
```diff
@@ -540,7 +540,7 @@ func (c *SimpleRESTClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
 	}
 
 	mapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
-	expander := restmapper.NewShortcutExpander(mapper, discoveryClient)
+	expander := restmapper.NewShortcutExpander(mapper, discoveryClient, nil)
 	return expander, nil
 }
```
[the remark this answers is not in the document]
Comment (a-hilaly) [reply pkg/kubernetes/client.go:545]: Looks like it's safe.. https://github.com/kubernetes/client-go/blob/master/restmapper/shortcut.go#L167-L169

[the remark this answers is not in the document]
Comment (a-hilaly) [reply pkg/kubernetes/client.go:545]: Maybe we can pass a function that will log the warning?

File: pkg/kubernetes/client.go [hunk 2]
```diff
@@ -540,7 +540,7 @@ func (c *SimpleRESTClientGetter) ToRESTMapper() (meta.RESTMapper, error) {
 	}
 
 	mapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
-	expander := restmapper.NewShortcutExpander(mapper, discoveryClient)
+	expander := restmapper.NewShortcutExpander(mapper, discoveryClient, nil)
```
Comment (yuxiang-zhang) [inline pkg/kubernetes/client.go:543]:
```suggestion
	expander := restmapper.NewShortcutExpander(mapper, discoveryClient, func(a string) {
		logger.Warning("unexpected warning message %s", a)
	})
```

Comment (a-hilaly) [reply pkg/kubernetes/client.go]: That's great!

**Review by a-hilaly — APPROVED** [review]
Thank you @yuxiang-zhang !

**Pull request merged**